pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+					folderUID string
+				}{
+					{name: "at root", folderUID: ""},
+					{name: "in folder", folderUID: ctx.TestFolder.UID},
+				}
+
+				for _, loc := range locations {
+					t.Run(loc.name, func(t *testing.T) {
+						if roleCapabilities.canCreate {
+							// Test can create dashboard
+							dash, err := createDashboard(t, identity.Client, identity.Name+" Dashboard "+loc.name, &loc.folderUID, nil)
+							require.NoError(t, err)
+							require.NotNil(t, dash)
+
+							// Verify if dashboard was created in the correct folder
+							if loc.folderUID != "" {
+								spec := dash.Object["spec"].(map[string]interface{})
+								require.Equal(t, loc.folderUID, spec["folderUID"])
+							}
+
+							// Clean up
+							err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+							require.NoError(t, err)
+						} else {
+							// Test cannot create dashboard
+							_, err := createDashboard(t, identity.Client, identity.Name+" Dashboard "+loc.name, nil, nil)
+							require.Error(t, err)
+						}
+					})
+				}
+			})
+
+			// Test dashboard updates
+			t.Run("dashboard update", func(t *testing.T) {
+				// Create a dashboard with admin
+				dash, err := createDashboard(t, adminClient, "Dashboard to Update by "+identity.Name, nil, nil)
+				require.NoError(t, err)
+				require.NotNil(t, dash)
+
+				if roleCapabilities.canUpdate {
+					// Test can update dashboard
+					updatedDash, err := updateDashboard(t, identity.Client, dash, "Updated by "+identity.Name)
+					require.NoError(t, err)
+					require.NotNil(t, updatedDash)
+
+					// Verify the update
+					updatedSpec := updatedDash.Object["spec"].(map[string]interface{})
+					require.Equal(t, "Updated by "+identity.Name, updatedSpec["title"])
+				} else {
+					// Test cannot update dashboard
+					_, err := updateDashboard(t, identity.Client, dash, "Updated by "+identity.Name)
+					require.Error(t, err)
+				}
+
+				// Clean up
+				err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+				require.NoError(t, err)
+			})
+
+			// Test dashboard deletion permissions
+			t.Run("dashboard deletion", func(t *testing.T) {
+				// Create a dashboard with admin
+				dash, err := createDashboard(t, adminClient, "Dashboard for deletion test by "+identity.Name, nil, nil)
+				require.NoError(t, err)
+				require.NotNil(t, dash)
+
+				// Attempt to delete
+				err = identity.Client.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+				if roleCapabilities.canDelete {
+					require.NoError(t, err, "Should be able to delete dashboard")
+				} else {
+					require.Error(t, err, "Should not be able to delete dashboard")
+					// Clean up with admin if the test identity couldn't delete
+					err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+					require.NoError(t, err)
+				}
+			})
+
+			// TODO: Check if vieweing permission can be revoked as well.
```

Comment:
it can be - we can, from a specific dashboard, remove the ability for viewers to view that dashboard (or editors to edit it, etc)

Reply:
ah i see that's covered later on :) for this todo, are you meaning if a user is given no role?

Reply:
@marcoabreu you either give access or no access. given a user with a none role. you can scope a read to a  dashboard. there is no way to subtract permissions from any roles/permissions.

We only have additive permissions

Reply:
Got it, thanks!